server/MembraneDetection/MembraneDetection.py

```diff
@@ -0,0 +1,143 @@
+import histomicstk.preprocessing.color_deconvolution as htk_cdeconv
+import histomicstk.filters.shape as htk_shape_filters
+
+import numpy as np
+import scipy as sp
+
+import skimage.color
+import skimage.filters
+import skimage.io
+import skimage.measure
+import skimage.morphology
+
+from ctk_cli import CLIArgumentParser
+
+import logging
+logging.basicConfig()
+
+
+stain_color_map = {
+    'hematoxylin': [0.65, 0.70, 0.29],
+    'eosin':       [0.07, 0.99, 0.11],
+    'dab':         [0.27, 0.57, 0.78],
+    'null':        [0.0, 0.0, 0.0]
+}
+
+
+def main(args):
+
+    #
+    # Read Input Image
+    #
+    print('>> Reading input image')
+
+    im_input = skimage.io.imread(args.inputImageFile)[:, :, :3]
+    size_min = min(im_input.shape[0], im_input.shape[1])
+    im_input = im_input[0:size_min, 0:size_min, :]
+
+    #
+    # Perform color deconvolution -unnormalized
+    #
+    print('>> Performing color deconvolution')
+
+    stain_color_1 = stain_color_map[args.stain_1]
+    stain_color_2 = stain_color_map[args.stain_2]
+    stain_color_3 = stain_color_map[args.stain_3]
+
+    w = np.array([stain_color_1, stain_color_2, stain_color_3]).T
+
+    im_stains = htk_cdeconv.color_deconvolution(im_input, w).Stains
+
+    im_membrane_stain = im_stains[:, :, 2].astype(np.float)
+
+    #
+    # Perform membrane enhancement
+    #
+    print('>> Performing membrane enhancement')
+
+    # membrane filtering
+    im_deviation, im_thetas = htk_shape_filters.membranefilter(
+        im_membrane_stain, [1, 2, 3], 4, 2
```

Comment:
We need to expose sigma's, beta, and c as parameters to the CLI.

For sigmas, you can ask for sigma range (min_sigma, max_sigma) and the number of steps between min and max values. Then you can use numpy's linspace function to generate the sigma vector.